### PR TITLE
[ base ] implement `decEq` for `SnocList`

### PR DESCRIPTION
**Pull request by joelberkeley**

### CODE REVIEW

File: libs/base/Decidable/Equality.idr [hunk 1]
```diff
@@ -111,6 +111,17 @@ public export
...
 
+--------------------------------------------------------------------------------
+-- SnocList
+--------------------------------------------------------------------------------
+
+public export
+DecEq a => DecEq (SnocList a) where
+  decEq Lin Lin = Yes Refl
+  decEq (xs :< x) Lin = No absurd
+  decEq [] (xs :< x) = No absurd
```
Comment (buzden) [inline libs/base/Decidable/Equality.idr:122]: Did you want `[<]` instead of `[]` here? Types should not match with this

Comment (joelberkeley) [reply libs/base/Decidable/Equality.idr]: yes, fixed. Is `[<]` the same as `Lin`?

File: libs/base/Data/SnocList.idr [hunk 2]
```diff
@@ -21,6 +21,14 @@ public export
 asList : SnocList type -> List type
 asList = (reverse . cast)
 
+export
+Uninhabited (Lin = x :< xs) where
+  uninhabited Refl impossible
+
+export
+Uninhabited (x :< xs = Lin) where
+  uninhabited Refl impossible
```
Comment (joelberkeley) [inline libs/base/Data/SnocList.idr:30]: i've made a rough guess where these and `Biinjective` should go

File: libs/base/Decidable/Equality.idr (hunk 1 above)
Comment (buzden) [reply libs/base/Decidable/Equality.idr]: > yes, fixed. Is `[<]` the same as `Lin`?

It is

File: libs/base/Data/SnocList.idr (hunk 2 above)
Comment (buzden) [reply libs/base/Data/SnocList.idr:30]: It probably in a good place, but you should include `Data.SnocList` in `Decidable.Equality` then

**Comment by gallais** [comment]
This seems sensible. Once the build failure is fixed, I'll be happy to merge it.

**Comment by joelberkeley** [comment]
whoops - sorry I assumed this was a global build issue that needed fixing elsewhere. I see you've picked this up now

**Pull request merged**